### PR TITLE
Initial work on jinia

### PR DESCRIPTION
**Pull request by gone**

### CODE REVIEW

File: sampleapp/config/settings/local.py [hunk 1]
```diff
@@ -72,13 +72,15 @@
 # https://django-debug-toolbar.readthedocs.io/en/latest/installation.html#internal-ips
 INTERNAL_IPS = ["127.0.0.1", "10.0.2.2"]
 
-
+"""
+Only use in some dev conditions
```
Comment (gone) [inline sampleapp/config/settings/local.py]: just kill this

File: sampleapp/templates/account/account_base.jinja [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{% from 'account/components/account_box.jinja' import account_box %}
+{% extends "base.jinja" %}
+{% set account_back %}
+{% block account_back %}
```
Comment (gone) [inline sampleapp/templates/account/account_base.jinja:4]: super clever

File: sampleapp/templates/account/components/account_box.jinja [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                     src="{{ static("img/logo.svg") }}"
+                     alt="logo">
+            </div>
+        {% else %}
+            <img class="mx-auto h-12 w-auto"
+                 src="{{ static("img/logo.svg") }}"
+                 alt="logo">
+        {% endif %}
+        <h2 class="mt-2 mb-4 text-center text-xl font-bold text-gray-900"
+            slot="title">{{ title }}</h2>
```
Comment (gone) [inline sampleapp/templates/account/components/account_box.jinja]: I think you can pull out the slot attribute

File: sampleapp/templates/account/password_reset.jinja [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    <p>
+        {% trans %}
+            Forgotten your password? Enter your e-mail address below, and we'll send you an e-mail allowing you to reset it.
+        {% endtrans %}
+    </p>
+    <form method="post"
+          action="{{ url('account_reset_password') }}"
+          class="password_reset">
+        {% csrf_token %}
+        {{ inputgroup(form) }}
```
Comment (gone) [inline sampleapp/templates/account/password_reset.jinja:23]: I think this should just be {{form}} - inputgroup is for a single field/lable/error

File: sampleapp/templates/account/password_reset_from_key.jinja [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        {{ url('account_reset_password') }}
+        <p>
+            {% trans %}
+                The password reset link was invalid, possibly because it has already been used.  Please request a <a href="{{ passwd_reset_url }}">new password reset</a>.
+            {% endtrans %}
+        </p>
+    {% else %}
+        <form method="post" action="{{ action_url }}">
+            {% csrf_token %}
+            {{ inputgroup(form) }}
```
Comment (gone) [inline sampleapp/templates/account/password_reset_from_key.jinja:26]: ditto here

File: sampleapp/templates/account/signup_form.jinja [hunk 6]
```diff
@@ -1,10 +1,11 @@
+{% from 'django/forms/inputgroup.jinja' import inputgroup %}
 <div class="form-wrapper gap-x-6 justify-between">
     {{ errors }}
     {% for field, errors in fields %}
         {% if field.name == 'first_name' or field.name == 'last_name' %}
-            {% include "django/forms/inputgroup.html" with field=field errors=errors layout_classes="w-5/12" %}
+            {{ inputgroup(field, errors, 'w-5/12') }}
```
Comment (gone) [inline sampleapp/templates/account/signup_form.jinja:6]: this is much cleaner.

File: sampleapp/config/jinja2.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            results[-1].append(
+                random.randrange(  # noqa
+                    floor(0.5 * previous_val), ceil(2 * previous_val)  # noqa
+                )  # noqa
+            )
+    return results
+
+
+options = {
+    "constants": {"csrf_cookie_name": "sampleapp"},
```
Comment (gone) [inline sampleapp/config/jinja2.py]: I think this is actually sampleapp_csrftoken

File: sampleapp/templates/components/menu.jinja [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+{% set default_button %}
+{{ button()}}
+{% endset %}
+{% set default_content %}
+<div role="menu"
+     aria-orientation="vertical"
+     aria-labelledby="user-menu-button"
+     tabindex="-1"></div>
+{% endset %}
+{% macro menu(button=default_button, content=default_content, button_style="text-gray-500 group bg-white rounded-md
```
Comment (gone) [inline sampleapp/templates/components/menu.jinja:11]: can you move style to a block too? I think will be easier to read

File: sampleapp/templates/current_time.jinja [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<div x-data>
```
Comment (gone) [inline sampleapp/templates/current_time.jinja:1]: way better than mine!

File: sampleapp/templates/django/forms/default.jinja [hunk 10]
```diff
@@ -0,0 +1,8 @@
+{% from 'django/forms/inputgroup.jinja' import inputgroup %}
```
Comment (gone) [inline sampleapp/templates/django/forms/default.jinja:1]: does this work? as a macro?

File: sampleapp/templates/django/forms/label.jinja [hunk 11]
```diff
@@ -0,0 +1,9 @@
+{% macro label(field, tag='label') %}
+    {% if use_tag %}
```
Comment (gone) [inline sampleapp/templates/django/forms/label.jinja]: I don't think this is defined

File: sampleapp/templates/django/forms/widgets/password.jinja [hunk 12]
```diff
@@ -1,4 +1,4 @@
-<span " x-data="{input: 'password'}">
+<span x-data="{input: 'password'}">
```
Comment (gone) [inline sampleapp/templates/django/forms/widgets/password.jinja:1]: :+1:

File: sampleapp/templates/header/end.jinja [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+       class="block px-4 py-2 text-sm text-gray-700"
+       role="menuitem"
+       tabindex="-1">
+        Sign out
+    </a>
+</div>
+{% endset %}
+<div class="hidden md:ml-4 md:flex-shrink-0 md:flex md:items-center">
+    <!-- Profile dropdown -->
+    {{ menu(button(button_for_menu), content_for_menu, "bg-white rounded-full flex text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500", "origin-top-right absolute right-0 mt-2 w-48 py-1 bg-white focus:outline-none") }}
```
Comment (gone) [inline sampleapp/templates/header/end.jinja]: can you do a set for the styling?

File: sampleapp/templates/header/mobile_menu.jinja [hunk 14]
```diff
@@ -24,12 +24,12 @@
         <div class="flex items-center px-4 sm:px-6">
             <div class="flex-shrink-0">
                 <img class="h-10 w-10 rounded-full"
-                     src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
-                     alt="">
+                     src="https://images.unsplash.com/photo-1601814933824-fd0b574dd592?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1112&q=80"
```
Comment (gone) [inline sampleapp/templates/header/mobile_menu.jinja:27]: lol

File: sampleapp/templates/index.jinja [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+    <button type="button"
+            class="bg-white border border-black px-4 py-2"
+            class="w"
+            hx-push-url="false"
+            hx-get="/random-chart/"
+            hx-target="#random-chart"
+            hx-swap='innerHTML'>
+        Refresh Chart
+    </button>
+    <template slot="header">
```
Comment (gone) [inline sampleapp/templates/index.jinja]: is this doing anything?

File: sampleapp/templates/django/forms/widgets/password.jinja [hunk 16]
```diff
@@ -1,4 +1,4 @@
...
     <input :type="input"
            name="{{ widget.name }}"
            {% if widget.value != None %} value="{{ widget.value|stringformat:'s' }}"{% endif %}
```
Comment (gone) [inline sampleapp/templates/django/forms/widgets/password.jinja]: There's a spelling error in this file as well, although it was there before you touched it - cusor -> cursor

File: .gitignore [hunk 17]
```diff
@@ -180,6 +180,9 @@ Temporary Items
 .directory
 
 
+### VS Code ###
+.vscode/
```
Comment (samamorgan) [inline .gitignore:184]: I'd recommend committing any VSCode launch/tasks/settings JSON in the repo. Many devs use VSCode and having some pre-baked setup for a repo is invaluable.

**Review by samamorgan — APPROVED** [review]

**Pull request closed without merging**